Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/StateSynchronization/SpatialLocalization/HostCoordinateLocalizationMechanism.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Experimental.SpatialAlignment.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    /// <summary>
+    /// Helper localization method here the host finds a QR code and gives it to the observer
+    /// </summary>
+    internal abstract class HostCoordinateLocalizationMechanism : LocalizationMechanismBase
+    {
+        private readonly object lockObject = new object();
+        private readonly List<Guid> guids = new List<Guid>();
+        private Task<ISpatialCoordinate> initializeBroadcasterCoordinateTask = null;
+        private TaskCompletionSource<string> observerCoordinateIdToLookFor = null;
+
+        /// <summary>
+        /// The spatial coordinate service for sub class to instantiate and this helper base to rely on.
+        /// </summary>
+        protected abstract ISpatialCoordinateService SpatialCoordinateService { get; }
+        
+        /// <summary>
+        /// The logic for the host to figure out which coordinate to use for localizing with observer.
+        /// </summary>
+        /// <param name="token">The token that first requested this host coordinate.</param>
+        /// <returns>The spatial coordinate.</returns>
+        protected virtual async Task<ISpatialCoordinate> GetHostCoordinateAsync(Guid token)
```

Comment:
>token [](start = 85, length = 5)

I don't understand the function of this token. Other than debug logging it's not used. What does the token actually represent?

Reply:
It's supposed to be use for the session, but I haven't found need for it just yet.